src/cluster/etcd/watchmanager/manager.go

```diff
@@ -140,8 +153,14 @@ func (w *manager) Watch(key string) {
 
 				// NB(cw) when we failed to create a etcd watch channel
 				// we do a get for now and will try to recreate the watch chan later
-				if err = w.updateFn(key, nil); err != nil {
-					logger.Error("failed to get value for key", zap.Error(err))
+				if !watchInitialized {
```

Comment:
The variable `watchInitialized` only controls whether to do a get request. It should be named appropriately to not be confused with a state when watch was successfully created. How about naming it `firstAttempt` (with inverse condition check) or something like that?